source/geometry_model/box.cc

```diff
@@ -260,6 +260,183 @@ namespace aspect
+                else
+                {
+                  if(temporary_surface[0][i] != temporary_surface[0][i-1])
+                  {
+                    temp_surface_x.push_back(temporary_surface[0][i]);
+                    temp_surface_y.push_back(temporary_surface[1][i]);
+                  }
+                }
+              }
+          }
+
+            double vector_size = temp_surface_x.size();
+            MPI_Bcast(&vector_size, 1, MPI_DOUBLE, 0, this->get_mpi_communicator());
+            MPI_Bcast(&temp_surface_x[0], vector_size, MPI_DOUBLE, 0, this->get_mpi_communicator());
+            MPI_Bcast(&temp_surface_y[0], vector_size, MPI_DOUBLE, 0, this->get_mpi_communicator());
+        }
+        else
+        {
+          for (unsigned int i=0; i<local_surface_height.size(); i++)
+              MPI_Ssend(&local_surface_height[i][0], local_surface_height[1].size(), MPI_DOUBLE, 0, 42, this->get_mpi_communicator());
+
+          double vector_size = 0;
+          MPI_Bcast(&vector_size, 1, MPI_DOUBLE, 0, this->get_mpi_communicator());
+          temp_surface_x.resize(vector_size);
+          temp_surface_y.resize(vector_size);
+          MPI_Bcast(&temp_surface_x[0], vector_size, MPI_DOUBLE, 0, this->get_mpi_communicator());
+          MPI_Bcast(&temp_surface_y[0], vector_size, MPI_DOUBLE, 0, this->get_mpi_communicator());
+        }
+
+        surface_x = temp_surface_x;
+        surface_y = temp_surface_y;
+    }
+
+    template <int dim>
+    double
+    Box<dim>::depth_including_mesh_deformation(const Point<dim> &position) const
+    {
+          // Compute the depth including free surface position, this only works in 2D
+          // The surface points do not always match position points, so we linearly
+          // Interpolate between the two nearest surface points.
+          // This function is sometimes called before the surface has been saved,
+          // so during timestep zero we call the normal depth function.
+          double depth_from_surface;
+          if(this->get_timestep_number() > 0)
+          {
+            double height = 0;
+            double x1 = 0; double x2 = 0; double y1 = 0; double y2 = 0;
+
+            // Loop through all surface points and find the nearest ones to the given position.
+            for(unsigned int s=0; s<surface_x.size();++s)
+            {
+              x2 = surface_x[s];
+              y2 = surface_y[s];
+
+              if(x2 > position[0])
+                break;
+
+              // If we aren't greater, then increase point1.
+              x1 = x2; y1 = y2;
+            }
+
+            // If we are at the edge of the model domain, use that y value. Otherwise interpolate.
+            if(position[0] >= surface_x[surface_x.size() - 1])
+              height = surface_y[surface_y.size() - 1];
+            else if (position[0] <= surface_x[0])
+              height = surface_y[0];
+            else
+              height = ((y2 - y1)/(x2-x1))*(position[0] - x1) + y1;
```

Comment:
This will do (and probably do it better :-) ): https://dealii.org/developer/doxygen/deal.II/classFunctions_1_1InterpolatedTensorProductGridData.html

Or if the data are actually equally spaced, then https://dealii.org/developer/doxygen/deal.II/classFunctions_1_1InterpolatedUniformGridData.html